ShadowDOM/explainer.md

```diff
@@ -0,0 +1,401 @@
+
+## Problem
+Declarative Shadow DOM doesn't have an equivalent to Imperative Shadow DOM’s `adoptedStylesheets` attribute, which is a gap in the currently exposed functionality. As a result, any complex styling that would normally be accomplished via shared stylesheets will either require JavaScript (which goes against the goal of markup-based shadow DOM) or will require a <link> tag referencing an external file.  
+
+Relying on JavaScript for styling is not ideal for DSD for several reasons: 
+* One of the main goals of DSD is to not rely on JavaScript [for performance and accessibility purposes](https://web.dev/articles/declarative-shadow-dom).
+* Adding stylesheets via script may cause an FOUC (Flash of Unstyled Content).
+* The current `adoptedStylesheets` attribute only supports Constructable Stylesheets, not inline stylesheets or stylesheets from <link> tags [(note that the working groups have recently decided to lift this restriction)](https://github.com/w3c/csswg-drafts/issues/10013#issuecomment-2165396092). 
+
+While referencing an external file via the <link> tag for shared styles in DSD works today [(and is currently recommended by DSD implementors)](https://web.dev/articles/declarative-shadow-dom#server-rendering_with_style), it is not ideal for several reasons: 
+* If the linked stylesheet has not been downloaded and parsed, there may be an FOUC.
+* External stylesheets are considered “render blocking”, and Google’s Lighthouse guidelines for high-performance web content recommends [using inline styles instead](https://developer.chrome.com/docs/lighthouse/performance/render-blocking-resources#how_to_eliminate_render-blocking_stylesheets).
+* Google’s Lighthouse guidelines recommend minimizing network requests for best performance. Stylesheets included via <link> tags are always external resources that may initiate a network request (note that the network cache mitigates this for repeated requests to the same file).
+
+This example shows how a developer might use DSD to initialize a shadow root without JavaScript. 
+
+```html
+  <article-card> 
+    <template shadowrootmode="open"> 
+       <style> 
+         :host { 
+          border: 1px solid #e0e0e0; 
+          } 
+       </style> 
+    </template> 
+  </article-card> 
+```
+While this approach is acceptable for a single component, a rich web application may define many `<template>` elements. Since pages often use a consistent set of visual styles, these `<template>` instances must each include `<style>` tags with duplicated CSS, leading to unnecessary bloat and redundancy.  
+
+This document explores several proposals that would allow developers to apply styles to DSD without relying on JavaScript and avoiding duplication. 
+
+## Goals 
```

Comment:
IMO explicit goals worth calling out are:
- Allow reuse of styles in markup-based shadow DOM without requiring JavaScript
- Allow reuse of styles in markup-based shadow DOM without requiring external network requests